python/lsst/ip/diffim/makeKernel.py

```diff
@@ -273,71 +268,77 @@ def getSelectSources(self, exposure, sigma=None, doSmooth=True, idFactory=None):
         ------
         RuntimeError
-            If ``candidateList`` is empty or contains incompatible types.
+            If ``candidateList`` is empty after sub-selection.
         """
         if candidateList is None:
-            candidateList = self.getSelectSources(scienceExposure, doSmooth=not preconvolved)
-
-        if len(candidateList) < 1:
-            raise RuntimeError("No candidates in candidateList")
-
-        listTypes = set(type(x) for x in candidateList)
-        if len(listTypes) > 1:
-            raise RuntimeError("Candidate list contains mixed types: %s" % [t for t in listTypes])
-
-        if not isinstance(candidateList[0], lsst.afw.table.SourceRecord):
-            try:
-                candidateList[0].getSource()
-            except Exception as e:
-                raise RuntimeError(f"Candidate List is of type: {type(candidateList[0])} "
-                                   "Can only make candidate list from list of afwTable.SourceRecords, "
-                                   f"measAlg.PsfCandidateF or other type with a getSource() method: {e}")
-            candidateList = [c.getSource() for c in candidateList]
-
-        candidateList = diffimTools.sourceToFootprintList(candidateList,
-                                                          templateExposure, scienceExposure,
-                                                          kernelSize,
-                                                          self.kConfig.detectionConfig,
-                                                          self.log)
-        if len(candidateList) == 0:
-            raise RuntimeError("Cannot find any objects suitable for KernelCandidacy")
-
-        return candidateList
+            candidateList = self.getSelectSources(reference, doSmooth=not preconvolved)
+            if len(candidateList) < 1:
+                raise RuntimeError("No kernel candidates after detection and measurement.")
+
+        bitmask = reference.mask.getPlaneBitMask(self.config.badMaskPlanes)
+        good = np.ones(len(candidateList), dtype=bool)
+        # Make all candidates have the same size footprint, based on kernelSize.
+        for i, candidate in enumerate(candidateList):
+            # Only use the brightest peak; the input should be pre-deblended!
+            peak = candidate.getFootprint().getPeaks()[0]
+            size = 2*kernelSize + 1  # ensure the resulting box is odd
+            bbox = lsst.geom.Box2I.makeCenteredBox(candidate.getCentroid(),
+                                                   lsst.geom.Extent2I(size, size))
+            boxFootprint = lsst.afw.detection.Footprint(lsst.afw.geom.SpanSet(bbox))
+            boxFootprint.addPeak(peak.getFx(), peak.getFy(), peak.getPeakValue())
+            candidate.setFootprint(boxFootprint)
+
+            # Reject footprints not contained in either image.
+            if not reference.getBBox().contains(bbox) or not convolved.getBBox().contains(bbox):
+                good[i] = 0
```

Comment:
Is it preferred to assign `0` or `False` instead? For readability sake.

Reply:
Good catch. `0` is `False` in a numpy array, but you're right that making it explicitly `False` is clearer.